src/Maestro/Maestro.Web/Api/v2019_01_16/Controllers/GoalController.cs

```diff
@@ -0,0 +1,96 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Threading.Tasks;
+using Maestro.Data;
+using Maestro.Data.Models;
+using Maestro.Web.Api.v2019_01_16.Models;
+using Microsoft.AspNetCore.ApiVersioning;
+using Microsoft.AspNetCore.ApiVersioning.Swashbuckle;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.KeyVault;
+using Microsoft.EntityFrameworkCore;
+using System.Net;
+
+namespace Maestro.Web.Api.v2019_01_16.Controllers
+{
+    [Route("goals")]
+    [ApiVersion("2019-01-16")]
+    public class GoalController : ControllerBase
+    {
+        protected readonly BuildAssetRegistryContext _context;
+        public GoalController(BuildAssetRegistryContext context)
+        {
+            _context = context;
+        }
+
+        /// <summary>
+        ///   Exposes methods to set goal time <see cref="Goal"/>
```

Comment:
This documentation is exposed in the swagger documentation API.  "see Goal" isn't going to mean much to that endpoint.  For example: https://maestro-prod.westus2.cloudapp.azure.com/swagger/ui/index.html#/Assets/Assets_ListAssets, that string comes from here.

"Exposes methods to set goal time" isn't a particularly helpful document.

"Sets a build time goal for a given repository in a channel.  This goal is used for generating build timing reports."  Or something like that.

And the parameters can use some love too.  ChannelName = "Channel target for build time (e.g. ".NET Core 5")", DefintionId = "Azure DevOps Pipeline definition id from http://dev.azure.com/dnceng", GoalTime = "Goal for maximum build duration, in minutes"

Reply:
Added comments like recommended.